packages/svelte/src/legacy/legacy-client.js

```diff
@@ -169,7 +176,14 @@ class Svelte4Component {
  * @returns {void}
  */
 export function run(fn) {
-	user_pre_effect(fn);
+	user_pre_effect(() => {
+		fn();
+		var effect = /** @type {import('#client').Effect} */ (active_effect);
+		// If the effect is immediately made dirty again, mark it as maybe dirty to emulate legacy behaviour
+		if ((effect.f & DIRTY) !== 0) {
+			set_signal_status(effect, MAYBE_DIRTY);
```

Comment:
Would be cool to warn in dev mode so people don't migrate towards effect and only then run into an infinite loop

Reply:
What should the warning say here?

Reply:
I added a warning but unsure of the exact messaging. Maybe @Rich-Harris has a clearer idea on the wording.

Reply:
message seems fine but it would be cool to include details of _where_ the statement is. see how we do it in e.g. `binding_property_non_reactive`

Reply:
How do we know the name of the reactive signal? It could be a binding, or a local variable or even a proxy signal.

Reply:
Ah sorry I got confused — morning brain — thought that this warning applied to `$:` rather than `run` in which case we can just pass the location information from the compiler to the runtime. But yeah that doesn't make sense in this context